addons/crm_livechat/tests/test_crm_lead.py

```diff
@@ -33,7 +33,9 @@ def test_crm_lead_creation_guest(self):
         # public: should not be set as customer
         channel = self.env['mail.channel'].create({
             'name': 'Chat with Visitor',
-            'channel_partner_ids': [(4, self.user_anonymous.partner_id.id)]
+            'channel_type': 'livechat',
+            'channel_partner_ids': [(4, self.user_anonymous.partner_id.id)],
+            'livechat_operator_id': self.env.user.partner_id.id
```

Comment:
I think the main issue is that these tests manually create the channel, which should ideally never be done.

See how it is done in `im_livechat`, it would be nice if the tests here also used `_open_livechat_mail_channel`

Reply:
I don't agree. What we are testing here is channel ``_convert_visitor_to_lead`` and ensuring public people are never set as lead customer_id. It is independent of how it was created.

Anyway why are you changing a test that is working ? Is the original fix broken in any way ?

Reply:
We are going to set a constraint on the group_public_id, which will only be used when channel_type is the 'channel'.  This will cause a restriction on the people who can create/modify the channel/chat/group/livechat. In our test cases, those crm tests will no longer work cause the channel_type is not specified properly.